pytorch_forecasting/data/timeseries.py

```diff
@@ -869,16 +869,18 @@ def __getitem__(self, idx: int) -> Tuple[Dict[str, torch.Tensor], torch.Tensor]:
             sequence_length >= self.min_prediction_length
         ), "Sequence length should be at least minimum prediction length"
         # determine prediction/decode length and encode length
-        decoder_length = min(
-            time[-1] - (self.min_prediction_idx - 1),
-            self.max_prediction_length,
-            sequence_length - self.min_encoder_length,
-        )
+        decoder_length = int(min(
+                    time[-1] - (self.min_prediction_idx - 1),
+                    self.max_prediction_length,
+                    sequence_length - self.min_encoder_length,
+                ))
+
         encoder_length = sequence_length - decoder_length
         assert (
             decoder_length >= self.min_prediction_length
         ), "Decoder length should be at least minimum prediction length"
         assert encoder_length >= self.min_encoder_length, "Encoder length should be at least minimum encoder length"
+        assert isinstance(encoder_length, int), f"encoder_length = {encoder_length}, must be of integer type"
```

Comment:
I suggest we do the assert in the `__init__` method. Let's do it for encoder length and decoder_length, minimum and maximum.